controllers/billingentity_email_cronjob.go

```diff
@@ -0,0 +1,112 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/multierr"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/appuio/control-api/mailsenders"
+	"github.com/prometheus/client_golang/prometheus"
+
+	billingv1 "github.com/appuio/control-api/apis/billing/v1"
+)
+
+// BillingEntityEmailCronJob periodically checks billing entities and sends notification emails if appropriate
+type BillingEntityEmailCronJob struct {
+	client.Client
+
+	Recorder record.EventRecorder
+	Scheme   *runtime.Scheme
+
+	MailSender mailsenders.MailSender
+
+	mailRecipientAddress string
+
+	failureCounter prometheus.Counter
+	successCounter prometheus.Counter
+}
+
+func NewBillingEntityEmailCronJob(client client.Client, eventRecorder record.EventRecorder, scheme *runtime.Scheme, mailSender mailsenders.MailSender, MailRecipientAddress string) BillingEntityEmailCronJob {
+	return BillingEntityEmailCronJob{
+		Client:               client,
+		Recorder:             eventRecorder,
+		MailSender:           mailSender,
+		mailRecipientAddress: MailRecipientAddress,
+		failureCounter:       newFailureCounter("control_api_billingentity_emails"),
+		successCounter:       newSuccessCounter("control_api_billingentity_emails"),
+	}
+
+}
+
+func (r *BillingEntityEmailCronJob) GetMetrics() prometheus.Collector {
```

Comment:
How hard would it be to have a test which verifies that the metrics are updated and reported correctly?